bahmni_atom_feed/models/order_save_service.py

```diff
@@ -384,7 +384,8 @@ def _create_sale_order_line_function(self, sale_order, order):
 
             if prod_lot != None:
                 life_date = prod_lot.life_date and datetime.strptime(prod_lot.life_date, DTF)
-                sale_order_line['price_unit'] = prod_lot.sale_price if prod_lot.sale_price > 0.0 else sale_order_line['price_unit']
+                if self.env.ref('bahmni_sale.sale_price_basedon_cost_price_markup').value == '1':
+                    sale_order_line['price_unit'] = prod_lot.sale_price if prod_lot.sale_price > 0.0 else sale_order_line['price_unit']
```

Comment:
is the "else" statement really required? we are just replacing with the same value from the existing sale_order_line['price_unit']

Reply:
Actually if we are assigning the sale price of the lot_id but if the price is 0.0 means not set then we cant get that value so we will make sure that we will not apply it. So whatever the price of the previously set by product will remain same.

Reply:
I understand that. I was commenting of need for 
> sale_order_line['price_unit'] = prod_lot.sale_price if prod_lot.sale_price > 0.0 else sale_order_line['price_unit']

this could be simply be
> sale_order_line['price_unit'] = prod_lot.sale_price if prod_lot.sale_price > 0.0; 

Right? The above says only if the lot sale price is > 0, update the sale_order_line.price_unit. The else code effectively does 
> sale_order_line['price_unit'] = sale_order_line['price_unit']

Reply:
sorry, please disregard the above comment. that style applies to ruby!